main.py

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/python3
+# main.py
+import sys
+from parser import parse
+from graph import Graph
+
+if len(sys.argv) != 2:
+    raise ValueError(
+        'Invalid Input, must be given a file which is in the CoNLL-U format')
+
+data = parse(open(sys.argv[1], 'r').read())
+print(data)
```

Comment:
I don't think the entire, possibly huge, data has to be printed. Instead, if there are cycles, you should print the entire sentence of data where the cycle was found.